### PR TITLE
Fix #9197, wrong esil for aarch64 ldr

### LINKED ISSUE #9197: ESIL: ldur is not working
_str_ and _ldr_ are working:
```
str x2, [sp + local_18h] 
ldr x0, [sp + local_18h]   
```

But in this combination only _stur_ works:
```
stur x0, [x29 - local_8h] 
ldur x1, [x29 - local_8h]         
```

_stur_ stores the value correctly, but _ldur_ won't touch x1, so x1 keeps it's value. I also checked that x0 != x1.

I noticed this in a couple of iOS binaries. Here is [one](https://github.com/samuelclay/NewsBlur/blob/55c36a24f38c25561e64b4365767547e3945ac1b/clients/ios/NewsBlur.ipa) from an open source project. The function is _"method.FirstTimeUserAddFriendsViewController.setFacebookButton:"_.

_______

> radare2 2.3.0-git 17047 @ darwin-x86-64 git.2.2.0-236-gbec01d21a
> commit: bec01d21ac735ca1b25842a27e3c2389d310871a build: 2018-01-15__01:01:25

### PR DESCRIPTION
**Pull request by giuscri**
See #9197.

Test: https://github.com/radare/radare2-regressions/pull/1186

Diff is messy. Actually, the change is small.

### CODE REVIEW

File: libr/anal/p/anal_arm_cs.c [hunk 1]
```diff
@@ -1022,53 +1022,51 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
 		default:
 		    break;
 		}
-		if ((int)MEMDISP64(1) < 0) {
```
Comment (giuscri) [inline libr/anal/p/anal_arm_cs.c]: I couldn't understand this if: why having a negative offset should result in an direct ESIL translation, while for positive offset you still have to check other things - see below?

Also, resulting expression is something like `dst,base,20,-,=[8]` which to my understanding translates to `dst = *(20-base)`

**Comment by radare** [comment]
Then rebase :p

**Comment by giuscri** [comment]
To make the diff cleaner? How? I'm already up to date with master

**Comment by giuscri** [comment]
Maybe I could split the change in 2, 3 commits to help diff and your review and you're squashing them while merging

**Comment by giuscri** [comment]
Actually passing `-b` to git diff ignores the change in whitespaces/indentation levels.
```
$ git diff upstream/master -b
diff --git a/libr/anal/p/anal_arm_cs.c b/libr/anal/p/anal_arm_cs.c
index 8bcbbc7a6..61df5545d 100644
--- a/libr/anal/p/anal_arm_cs.c
+++ b/libr/anal/p/anal_arm_cs.c
@@ -1022,10 +1022,6 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
                default:
                    break;
                }
-               if ((int)MEMDISP64(1) < 0) {
-                       r_strbuf_setf (&op->esil, "%s,%s,%"PFMT64d",-,=[%d]",
-                               REG64(0), MEMBASE64(1), -(int)MEMDISP64(1), size);
-               } else {
                if (ISMEM64(1)) {
                        if (HASMEMINDEX64(1)) {
                                if (LSHIFT2_64(1)) {
@@ -1039,6 +1035,9 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
                                if (LSHIFT2_64(1)) {
                                        r_strbuf_appendf (&op->esil, "%s,%d,%"PFMT64d",%s,+,[%d],%s,=",
                                                        MEMBASE64(1), LSHIFT2_64(1), MEMDISP64(1), DECODE_SHIFT64(1), size, REG64(0));
+                               } else if ((int)MEMDISP64(1) < 0){
+                                       r_strbuf_appendf (&op->esil, "%"PFMT64d",%s,-,DUP,tmp,=,[%d],%s,=,",
+                                                       -(int)MEMDISP64(1), MEMBASE64(1), size, REG64(0));
                                } else {
                                        r_strbuf_appendf (&op->esil, "%s,%"PFMT64d",+,DUP,tmp,=,[%d],%s,=,",
                                                        MEMBASE64(1), MEMDISP64(1), size, REG64(0));
@@ -1069,7 +1068,6 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
                                        IMM64(1), size, REG64(0));
                        }
                }
-               }
                break;
                }
        case ARM64_INS_CCMP:
```

**Pull request merged**